datamodel/src/main/kotlin/com/hartwig/actin/datamodel/molecular/PanelRecord.kt

```diff
@@ -16,7 +16,10 @@ data class PanelRecord(
     override val hasSufficientQuality: Boolean
 ) : MolecularTest {
 
-    override fun testsGene(gene: String): Boolean {
-        return testedGenes.contains(gene)
+    override fun testsGene(gene: String, molecularTestTargets: List<MolecularTestTarget>): Boolean {
+        if (molecularTestTargets.isEmpty()) {
+            return geneSpecifications[gene] != null
+        }
```

Comment:
I think a `Set` would be more suited here.

Also, current implementation is fine I think but maybe you could consider demanding the `molecularTestTargets` to include at least 1 entry or otherwise throw an exception. Every caller of this function should know the targets it is interested in.